### PR TITLE
Refactor using only javascript

### PR DESCRIPTION
**Pull request by louisinger**
*Huge breaking changes in this PR*

TL;DR
- delete Travis: replace by GH CI.
- Welcome to Typescript (and tsdx).
- Welcome to Yarn
- Remove emscripten and all C parts.
- reimplement blech32 with Pure Javascript (+ the [long](https://www.npmjs.com/package/long) package)

The lib exports two things:
- all the "base" functions from `src/blech32.ts` file.
- an easy-to-use `Blech32Address` class.

```typescript
const addr = Blech32Address.from(witness, blindingPubKey, 'el', 0).address // wrap the "encode" function.
const blindingPublicKey = Blech32Address.fromString("elqq....").blindingPublicKey // wrap the "decode" function.
```

Note that the tests vectors are the same than the previous one.

@tiero please review

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export * from "./blech32";
+export { Blech32Address } from "./blech32Addr";
```
Comment (tiero) [inline src/index.ts:2]: are you sure this works? I recall we may need to export all

**Comment by tiero** [comment]
cc @asoltys

File: src/index.ts (hunk 1 above)
Comment (louisinger) [reply src/index.ts:2]: it seems to work yes

File: src/blech32.ts [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+}
+
+function createChecksum(
+  hrp: string,
+  data: Uint8Array,
+  enc: EncodingType
+): number[] {
+  const values = Array.from(hrpExpand(hrp))
+    .concat(Array.from(data))
+    .concat([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]);
```
Comment (altafan) [inline src/blech32.ts]: maybe better to put this into a variable?

**Comment by altafan** [comment]
Even if test vectors haven't changed, they seem a bit poor. Can we add at least 2 other for p2wpkh and p2wsh mainnet?
Also, would be better to add failing tests to make sure only native segwit CT addresses can be parsed with this module.
An error should be expected if trying to parse any other type of CT address or invalid ones.

**Comment by louisinger** [comment]
> Even if test vectors haven't changed, they seem a bit poor. Can we add at least 2 other for p2wpkh and p2wsh mainnet?
> Also, would be better to add failing tests to make sure only native segwit CT addresses can be parsed with this module.
> An error should be expected if trying to parse any other type of CT address or invalid ones.

We could also use this: https://github.com/ElementsProject/elements/commit/8fced5304224a6f7a4084b71c14d44c4fa69634f

**Review by altafan — APPROVED** [review]

**Pull request merged**